### PR TITLE
pd: support dpa1

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
Summary of this PR:

1. upload DPA-1 related code
2. merge much develop code
3. add all eager composite operators except `softmax_grad`, `p_norm_grad`, `split_grad`, and `concat_grad` to the composite operator blacklist(<https://github.com/deepmodeling/deepmd-kit/pull/4414/files#diff-e678abb052b278f8a479f8d13b839a9ec0effd9923478a850bc13758f918e1e9R134-R148>) to significantly improve model execution speed (reducing the time taken from 100% more than PyTorch to about 10% to 15% more).

related PR: https://github.com/lanpa/tensorboardX/pull/728

### Training curve:

training_curves_comparison_eager_opt

### Accuracy test(left: paddle, right: torch):

image

Ralated optimization of Paddle framework:
- [x] https://github.com/PaddlePaddle/Paddle/pull/69349
- [x] https://github.com/PaddlePaddle/Paddle/pull/69333
- [x] https://github.com/PaddlePaddle/Paddle/pull/69479
- [x] https://github.com/PaddlePaddle/Paddle/pull/69515
- [x] https://github.com/PaddlePaddle/Paddle/pull/69487
- [x] https://github.com/PaddlePaddle/Paddle/pull/69661
- [x] https://github.com/PaddlePaddle/Paddle/pull/69660
- [x] https://github.com/PaddlePaddle/Paddle/pull/69596
- [x] https://github.com/PaddlePaddle/Paddle/pull/69556

## Release Notes

- **New Features**
	- Introduced several new classes for molecular descriptors, including `DescrptDPA1`, `DescrptBlockSeAtten`, and `LayerNorm`, enhancing the modeling capabilities for molecular simulations.
	- Added new JSON configuration files for model parameters and multitask models related to water simulations.
	- Implemented new test classes for validating the functionality of the `DPAtomicModel` and various descriptor classes.
	- Added new test classes for evaluating denoising models, including `TestDenoiseModelDPA1` and `TestDenoiseModelDPA2`.
	- Enhanced the `ModelWrapper` class to clarify the handling of model parameters and state management.

- **Bug Fixes**
	- Improved internal logic for handling model state saving and loading, ensuring consistency in outputs.

- **Documentation**
	- Enhanced type hints and return annotations across various classes and methods for better clarity.

- **Tests**
	- Expanded the testing framework with new test cases for denoising models and descriptor functionalities, ensuring robust validation of features.
	- Activated previously skipped tests for energy models, improving test coverage.
	- Enhanced multitask training tests with new configuration handling and test classes.

### CODE REVIEW

File: deepmd/pd/train/training.py [hunk 1]
```diff
@@ -66,6 +63,7 @@
     make_stat_input,
 )
 from deepmd.pd.utils.utils import (
+    nvprof_context,
```
[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply deepmd/pd/train/training.py:66]: nvprof_context is still in deepmd/pd/utils/utils.py

**Review by njzjz — COMMENTED** [review]
Please add consistent tests. (please note my changes in #4438)

**Comment by HydrogenSulfate** [comment]
> Please add consistent tests. (please note my changes in #4438)

Thanks for comments, and updated 2 consistency UT of dpa1, 
=========== 346 passed, 446 skipped, 1 warning in 48.38s =============

**Comment by HydrogenSulfate** [comment]
@iProzd PR of pd: DPA-1 and [pd: DPA-2](https://github.com/deepmodeling/deepmd-kit/pull/4418) is ready for review.

File: deepmd/pd/train/wrapper.py [hunk 2]
```diff
@@ -64,74 +64,13 @@ def __init__(
...
-                        link_class = self.model[model_key_link].__getattr__(
-                            class_type_link
-                        )
-                        link_class.share_params(
-                            base_class, shared_level_link, resume=resume
-                        )
-                        log.warning(
-                            f"Shared params of {model_key_base}.{class_type_base} and {model_key_link}.{class_type_link}!"
-                        )
+        raise NotImplementedError("share_params is not implemented yet")
```
[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply deepmd/pd/train/wrapper.py:73]: > Why is share_params not supported ?

I remove them for code coverage as this method seems not used currently in  water/se_e2_a/se_atten/DPA-2 with paddle backend.

[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply deepmd/pd/train/wrapper.py:73]: > Why is share_params not supported ?

I can re-add share_params if necessary...

[the remark this answers is not in the document]
Comment (iProzd) [reply deepmd/pd/train/wrapper.py]: These are tested in source/tests/pt/test_multitask.py, maybe it's better to add this test in pd.

[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply deepmd/pd/train/wrapper.py]: > These are tested in `source/tests/pt/test_multitask.py`, maybe it's better to add this test in pd.

I think I might have mistakenly attributed another randomly occurring CI CUDA issue(#4439) to the `test_muktitask.py` file. I reviewed the file again to ensure it can correctly pass the unit tests.

**Comment by njzjz** [comment]
@coderabbitai resolve

File: source/tests/pd/test_multitask.py [hunk 3]
```diff
@@ -39,13 +40,20 @@
     with open(multitask_template_json) as f:
         multitask_template = json.load(f)
 
+    global multitask_sharefit_template
+    multitask_sharefit_template_json = str(
+        Path(__file__).parent / "water/multitask_sharefit.json"
+    )
+    with open(multitask_sharefit_template_json) as f:
+        multitask_sharefit_template = json.load(f)
```
[the remark this answers is not in the document]
Comment (njzjz) [reply source/tests/pd/test_multitask.py:48]: @HydrogenSulfate could you check this notice?

[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply source/tests/pd/test_multitask.py]: > @HydrogenSulfate could you check this notice?

Thanks, this global variable is used in sharefit UT, which is not related to this PR, now removed.

**Review by iProzd — APPROVED** [review]

**Review by njzjz — APPROVED** [review]
